editor/editor_system.cpp

```diff
@@ -782,6 +787,87 @@ Variant EditorSystem::saveAll(const Event* event)
 	return true;
 }
 
+bool EditorSystem::copySceneIndependentFiles(String& exportBase)
+{
+	Vector<Pair<String, String>> execFiles = {
+		{ "build/game/Release/Game.exe",
+		    "Game.exe" },
+		{ "build/game/Release/alut.dll",
+		    "alut.dll" },
+		{ "build/game/Release/OpenAL32.dll",
+		    "OpenAL32.dll" },
+		{ "build/game/Release/wrap_oal.dll",
+		    "wrap_oal.dll" },
+		{ "rootex.root", "rootex.root" },
+		{ "rootex/vendor/Debugger/Debugger.lua", "rootex/vendor/Debugger/Debugger.lua" },
+		{ "game/startup.lua", "game/startup.lua" },
+		{ "rootex/vendor/ASSAO/ASSAO.hlsl", "rootex/vendor/ASSAO/ASSAO.hlsl" },
+		{ "rootex/vendor/Middleclass/Middleclass.lua", "rootex/vendor/Middleclass/Middleclass.lua" },
+	};
+	for (auto& file : execFiles)
+	{
+		if (OS::IsExists(file.first))
+		{
+			OS::RelativeCopyFile(file.first, exportBase + file.second);
+		}
+		else
+		{
+			WARN("Export failed, " + file.first + " not found");
+			return false;
+		}
+	}
+	return true;
+}
+
+Variant EditorSystem::exportScene(const Event* event)
+{
+	saveAll(nullptr);
+
+	Scene* toExportScene = SceneLoader::GetSingleton()->getCurrentScene();
+	String sceneName = toExportScene->getName();
+
+	String currExportDir;
+	int i = 0;
+	do
+	{
+		currExportDir = "exports/" + sceneName;
+		if (i != 0)
+		{
+			currExportDir += std::to_string(i);
+		}
+		currExportDir += "/";
+		i++;
+	} while (OS::IsExists(currExportDir));
+
+	OS::CreateDirectoryName(currExportDir);
+
+	if (!copySceneIndependentFiles(currExportDir))
+	{
+		OS::DeleteDirectory(currExportDir);
+		return false;
+	}
+
+	OS::CreateDirectoryName(currExportDir + "game/assets/");
+	OS::CreateDirectoryName(currExportDir + "rootex/assets/");
+	OS::RelativeCopyFolder(String("game/assets/"), currExportDir + "game/assets/");
```

Comment:
The extra `String()` will not be required if the parameter used `const String&`. Also this is not allowed according to standard, but MSVC allows this